cluster-operator/src/test/java/io/strimzi/operator/cluster/model/EntityOperatorTest.java

```diff
@@ -471,4 +474,347 @@ public void testImagePullPolicy() {
     public static void cleanUp() {
         ResourceUtils.cleanUpTemporaryTLSFiles();
     }
+
+    @Test
+    public void testTopicOperatorContainerEnvVars() {
+
+        ContainerEnvVar envVar1 = new ContainerEnvVar();
+        String testEnvOneKey = "TEST_ENV_1";
+        String testEnvOneValue = "test.env.one";
+        envVar1.setName(testEnvOneKey);
+        envVar1.setValue(testEnvOneValue);
+
+        ContainerEnvVar envVar2 = new ContainerEnvVar();
+        String testEnvTwoKey = "TEST_ENV_2";
+        String testEnvTwoValue = "test.env.two";
+        envVar2.setName(testEnvTwoKey);
+        envVar2.setValue(testEnvTwoValue);
+
+        List<ContainerEnvVar> testEnvs = new ArrayList<>();
+        testEnvs.add(envVar1);
+        testEnvs.add(envVar2);
+
+        ContainerTemplate topicOperatorContainer = new ContainerTemplate();
+        topicOperatorContainer.setEnv(testEnvs);
+
+        Kafka resource =
+                new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image, healthDelay, healthTimeout))
+                        .editSpec()
+                            .withNewEntityOperator()
+                                .withTopicOperator(entityTopicOperatorSpec)
+                                .withUserOperator(entityUserOperatorSpec)
+                                .withNewTemplate()
+                                    .withTopicOperatorContainer(topicOperatorContainer)
+                                .endTemplate()
+                            .endEntityOperator()
+                        .endSpec()
+                        .build();
+
+        EntityOperator eo = EntityOperator.fromCrd(resource, VERSIONS);
+        EntityTopicOperator eto = eo.getTopicOperator();
+
+        List<EnvVar> containerEnvVars = eto.getEnvVars();
+
+        int keyCount = 0;
+
+        for (EnvVar envVar : containerEnvVars) {
+
+            if (envVar.getName().equals(testEnvOneKey) || envVar.getName().equals(testEnvTwoKey)) {
+                if (envVar.getValue().equals(testEnvOneValue) || envVar.getValue().equals(testEnvTwoValue)) {
+                    keyCount++;
```

Comment:
I fonder is comething like this would be better for testing the envvars in all these tests:

```
assertTrue(cont.getEnv().stream().filter(env -> "TEST_ENV_1".equals(env.getName())).map(EnvVar::getValue).findFirst().orElse("").equals("test.env.one"));
assertTrue(cont.getEnv().stream().filter(env -> "TEST_ENV_2".equals(env.getName())).map(EnvVar::getValue).findFirst().orElse("").equals("test.env.two"));
```

It will be easier to read and will not just rely on a counter.

Reply:
Agreed, I have refactored all the tests